source/common/config/grpc_mux_impl.cc

```diff
@@ -206,5 +220,47 @@ void GrpcMuxImpl::handleEstablishmentFailure() {
   }
 }
 
+void GrpcMuxImpl::populateControlPlaneInfo(const envoy::api::v2::DiscoveryResponse& message) {
+  if (message.has_control_plane()) {
+    Protobuf::util::MessageDifferencer message_differencer;
+    envoy::admin::v2alpha::ControlPlaneConfigDump::ConfigSourceControlPlaneInfo
+        config_source_control_plane_info;
+    bool config_exists = false;
+    for (auto& config_control_plane :
+         GrpcMuxImpl::per_service_control_plane_info_[server_type_url_]) {
+      if (message_differencer.Compare(grpc_service_, config_control_plane.grpc_service())) {
+        config_source_control_plane_info = config_control_plane;
+        config_exists = true;
+        break;
+      }
+    }
+    config_source_control_plane_info.mutable_control_plane()->set_identifier(
```

Comment:
Can you add a few more comments to this method? I think it's correct but it took me a while to figure out the different cases.